### PR TITLE
OXT-1594: Forward Seal according to /config/config.pcrs

### PR DESCRIPTION
**Pull request by jandryuk**
This PR is groundwork for OXT-1594.  It makes `seal-system` forward seal according to /config/config.pcrs.  It does not change the pcr selection.

It is now flexible and will seal against any selection of PCRs.  If the selection set is empty, then seal-system fails (`seal-system: forward seal of key failed`) since tpm2_sealdata fails (`ERROR: Unable to run tpm2_sealdata`).  We can also drop out PCR 15 which has the rootfs measurement.  Is that too flexible?

There are also some other cleanups I noticed along the way.

### CODE REVIEW

**Comment by eric-ch** [comment]
Build: [$6556](http://openxt-builder.ainfosec.com:8010/builders/openxt64/builds/6556)

File: recipes-openxt/xenclient-tpm-scripts/xenclient-tpm-scripts/tpm-functions [hunk 1]
```diff
@@ -25,6 +25,7 @@
 
 clean_old_tpm_files () {
     [ -e /boot/system/tpm/config.tss ] && rm /boot/system/tpm/config.tss
+    [ -e /boot/system/tpm/config.tss.pcrs ] && rm /boot/system/tpm/config.tss.pcrs
```
[the remark this answers is not in the document]
Comment (jandryuk) [reply recipes-openxt/xenclient-tpm-scripts/xenclient-tpm-scripts/tpm-functions]: Yes, `rm -f` would be better, but I matched the existing form.

**Comment by eric-ch** [comment]
Will merge soon.

@crogers1 Is this something you want in stable-9? The ticket mentions only 9.0.0 as a target, but I am guessing you may want to consider it.

**Comment by jandryuk** [comment]
@crogers1, I would like this in stable-9, but I understand if you don't take it.

Currently, a forward-seal uses PCRs 0-8, 15, & 17-19 regardless of what is specified in /config/config.pcrs.  This happened in 9e4f5c9f84bb76f1ebdc6a01cf6dea5d28362cb7.  Before that, /config/config.pcrs was followed properly.

I tagged it with OXT-1594 since it's preparatory work for that.  But you can also consider it a standalone bug independent of tweaking a PCR selection per fixing docks in OXT-1594.

**Comment by crogers1** [comment]
How is the upgrade case handled?  I see that unsealdata will use the same pcrs dumped by forward-seal, but if we wanted to add/remove pcrs from the measurement, how would that be handled by the upgrade?

**Comment by jandryuk** [comment]
https://github.com/OpenXT/installer/blob/master/part2/stages/Functions/install-main#L451

`seal_system()` in the installer calls `write_config_pcrs` to update `/config/config.pcrs`and then runs `seal-system -f` which will use the new selection.

**Comment by crogers1** [comment]
Ah good point I forgot about that line.  Ok yea I'm good with this in stable-9

**Pull request merged**